### PR TITLE
[Pages] supported versions

### LINKED ISSUE #3759: Pages Docs: Add detailed information on build image
### Expected Behavior

Be explicit about the build system environment / build system image (most importantly, OS version) in the documentation, as well as what software is available for choosing.

### Actual Behavior

I once specified `PYTHON_VERSION=3.9` as an environment variable for one of my Pages sites, which ended up with the following error message:

```text
Installing dependencies
/opt/build/bin/run-build-functions.sh: line 168: /opt/buildhome/python3.9/bin/activate: No such file or directory
Error setting python version from the PYTHON_VERSION environment variable
Please see https://github.com/netlify/build-image/#included-software for current versions
Failed: build command exited with code: 1
```

Furthermore, the link in the message is broken. It took me multiple tries to figure out the exact build environment and the correct documentation ([netlify/build-image@`xenial`/included_software.md](https://github.com/netlify/build-image/blob/xenial/included_software.md)).

### Section that requires update

<https://developers.cloudflare.com/pages/platform/build-configuration/#language-support-and-tools>

### Additional information

Original thread: [Feature Request: Publish Cloudflare Pages build system version, and add a newer system for choosing - Cloudflare Community](https://community.cloudflare.com/t/feature-request-publish-cloudflare-pages-build-system-version-and-add-a-newer-system-for-choosing/368392)

### PR DESCRIPTION
**Pull request by deadlypants1973**
closes #3759

### CODE REVIEW

File: content/pages/platform/build-configuration.md [hunk 1]
```diff
@@ -108,3 +108,107 @@ Many common tools have been pre-installed as well. The environment variable avai
...
+
+* Erlang
+    21 (default)
+
+* Elixir
+    1.7 (default)
+
+* Swift - `SWIFT_VERSION`, `.swift-version`
+    Not installed by default.
+    Supports any version that swiftenv can install later than `4.x`. Versions `4.x` and earlier will not work due to incompatible shared libraries.
```
Comment (WalshyDev) [inline content/pages/platform/build-configuration.md]: it actually is installed by default

File: content/pages/platform/build-configuration.md [hunk 2]
```diff
@@ -108,3 +108,107 @@ Many common tools have been pre-installed as well. The environment variable avai
...
+
+* Hugo - `HUGO_VERSION`
+    0.54 extended (default)
+    Any version installable via `binrc`.
+
+Gutenburg - `GUTENBERG_VERSION`
+    Any version installable via `binrc`.
+
+* Zola - `ZOLA_VERSION`
+    Any version installable via binrc.
```
Comment (WalshyDev) [inline content/pages/platform/build-configuration.md]: Max version is 0.14.0
Min version is 0.5.0

File: content/pages/platform/build-configuration.md [hunk 3]
```diff
@@ -108,3 +108,107 @@ Many common tools have been pre-installed as well. The environment variable avai
...
+    3.81
+
+* OptiPNG
+    0.6.4
+
+* Doxygen
+    1.8.6
+
+* Homebrew 
+    Early alpha.
```
Comment (WalshyDev) [inline content/pages/platform/build-configuration.md]: Don't think we have homebrew

**Review by kodster28 — APPROVED** [review]

**Comment by Lar1o87** [comment]
No​ I don't want yours

ເມື່ອ ອ., 19 ມ.ສ. 2022, 22:05 Kody Jackson ***@***.***>
ຂຽນວ່າ:

File: content/pages/platform/build-configuration.md [hunk 4]
```diff
@@ -73,38 +73,46 @@ If you need to use a specific version of a language, (for example, Node.js or Ru
...
-| PHP       | 5.6             | `PHP_VERSION`        |                           |
-| Python    | 2.7             | `PYTHON_VERSION`     | `runtime.txt`, `Pipfile`  |
-| Ruby      | 2.6.2           | `RUBY_VERSION`       | `.ruby-version`           |
-| .NET      | 3.1.302         |                      |                           |
+{{<table-wrap>}}
+
+| Language  | Default version | Supported versions | Environment variable             | File                     |
+| --------- | --------------- | ------------------ |--------------------------------- | -------------------------|
+| Clojure   |                 |                    |                                  |                          |
+| Elixir    | 1.7             | 1.7 (default) only.|                                  |                          |
```
Comment (iBug) [inline content/pages/platform/build-configuration.md:81]: What's the point of adding a `(default)` mark when there's a dedicated column for "Default version"?

File: content/pages/platform/build-configuration.md [hunk 5]
```diff
@@ -73,38 +73,46 @@ If you need to use a specific version of a language, (for example, Node.js or Ru
...
+| Erlang    | 21              | 21 (default) only. |                                  |                          |
+| Go        | 1.14.4          | 1.14.4 (default) and any version available on the Go downloads page. |`GO_VERSION`|                           |
+| Java      | 8               | 8 (default) only.  |                                  |                           |
+| Node.js   | 12.18.0         | 12.18.0 (default) and any version up to 17.x. |`NODE_VERSION`       | `.nvmrc`, `.node-version` |
+| PHP       | 5.6             | 5.6 (default), 7.2, 7.4 only.                         |`PHP_VERSION`        |                           |
+| Python    | 2.7             | 2.7 (default), 3.5, 3.7 only.                         |`PYTHON_VERSION`     | `runtime.txt`, `Pipfile`  |
+| Ruby      | 2.6.2           | Any version between 2.6.2 (default) and 2.7.5.        |`RUBY_VERSION`       | `.ruby-version`           |
+| Swift     | 5.2             | Installed by default. Pages supports any 5.x version. |`SWIFT_VERSION`| `.swift-version` |
+| .NET      | 3.1.302         | |                     |                           |
```
Comment (WalshyDev) [inline content/pages/platform/build-configuration.md:91]: Formatted table nicely and made some of the text in supported versions shorter and more consistent.

```suggestion
| Language  | Default version | Supported versions                  | Environment variable  | File                      |
| --------- | --------------- | ----------------------------------- | --------------------- | ------------------------- |
| Clojure   |                 |                                     |                       |                           |
| Elixir    | 1.7             | 1.7 only                            |                       |                           |
| Emacs     | 25              | 25 only                             |                       |                           |
| Erlang    | 21              | 21 only                             |                       |                           |
| Go        | 1.14.4          | Any version                         | `GO_VERSION`          |                           |
| Java      | 8               | 8 only                              |                       |                           |
| Node.js   | 12.18.0         | Any version up to 17.x              | `NODE_VERSION`        | `.nvmrc`, `.node-version` |
| PHP       | 5.6             | 5.6, 7.2, 7.4 only                  | `PHP_VERSION`         |                           |
| Python    | 2.7             | 2.7, 3.5, 3.7 only                  | `PYTHON_VERSION`      | `runtime.txt`, `Pipfile`  |
| Ruby      | 2.6.2           | Any version between 2.6.2 and 2.7.5 | `RUBY_VERSION`        | `.ruby-version`           |
| Swift     | 5.2             | Any 5.x version                     | `SWIFT_VERSION`       | `.swift-version`          |
| .NET      | 3.1.302         |                                     |                       |                           |
```

File: content/pages/platform/build-configuration.md [hunk 6]
```diff
@@ -73,38 +73,46 @@ If you need to use a specific version of a language, (for example, Node.js or Ru
...
+| GNU Make    | Version 3.8.1                               |                       |                     |
+| ImageMagick | Version 6.7.7                               |                       |                     |
+| jq          | Version 1.5                                 | |                     |
+| Leiningen   |                                             | |                     |
+| OptiPNG     | Version 0.6.4                               | |                     |
+| NPM         | Corresponds with Node.js version.| Any version available via npm. |`NPM_VERSION`        |
+| pip         | Version corresponding with Python version.  | |                     |
+| Pipenv      | Latest version.                             | |                     |
+| Yarn        | Version 1.22.10                             | Any version available via the Yarn installer. |`YARN_VERSION`|
+| Zola        | (requires environment variable)             | Any version from 0.5.0 to 0.14.0. Any version installable via `binrc`. |`ZOLA_VERSION`       |
```
Comment (WalshyDev) [inline content/pages/platform/build-configuration.md:116]: Same as above (I believe we do have a max on yarn but don't remember so we'll update when it pops up)

```suggestion
| Tools       | Default version                             | Supported versions               | Environment variable |
| ----------- | --------------------------------------------| -------------------------------- | -------------------- |
| Boot        | 2.5.2                                       | 2.5.2                            |                      |
| Bower       |                                             |                                  |                      |
| Cask        |                                             |                                  |                      |
| Composer    |                                             |                                  |                      |
| Doxygen     | Version 1.8.6                               |                                  |                      |
| Emacs       | Version 25                                  |                                  |                      |
| Gutenberg   | (requires environment variable)             | Any version                      | `GUTENBERG_VERSION`  |
| Hugo        | Version 0.54                                | Any version                      | `HUGO_VERSION`       |
| GNU Make    | Version 3.8.1                               |                                  |                      |
| ImageMagick | Version 6.7.7                               |                                  |                      |
| jq          | Version 1.5                                 |                                  |                      |
| Leiningen   |                                             |                                  |                      |
| OptiPNG     | Version 0.6.4                               |                                  |                      |
| NPM         | Corresponds with Node.js version            | Any version                      | `NPM_VERSION`        |
| pip         | Version corresponding with Python version   |                                  |                      |
| Pipenv      | Latest version                              |                                  |                      |
| Yarn        | Version 1.22.10                             | Any version                      | `YARN_VERSION`       |
| Zola        | (requires environment variable)             | Any version from 0.5.0 to 0.14.0 | `ZOLA_VERSION`       |
```

**Pull request merged**